pkgs/applications/networking/p2p/transmission/4.nix

```diff
@@ -141,8 +142,14 @@ stdenv.mkDerivation rec {
     EOF
   '';
 
+  passthru.tests = {
+    apparmor = nixosTests.transmission_4; # starts the service with apparmor enabled
+    smoke-test = nixosTests.bittorrent;
+  };
+
   meta = {
     description = "A fast, easy and free BitTorrent client";
+    mainProgram = (transmission.override { inherit enableQt enableGTK3 enableCli; }).meta.mainProgram;
```

Comment:
We should not couple expressions for 4.x and 3.x to allow for independent evolution.